p2p/connection.go

```diff
@@ -310,7 +315,13 @@ FOR_LOOP:
 			c.Logger.Debug("Send Ping")
 			wire.WriteByte(packetTypePing, c.bufWriter, &n, &err)
 			c.sendMonitor.Update(int(n))
-			c.flush()
+			// should be c.flush
+			go c.flush()
```

Comment:
why goroutine?

Reply:
ah I think TestPingPongTimeout would block on `c.flush()`

Reply:
you think?) does it mean you are not sure?

Reply:
running the test it hangs if it's `c.flush()`

Reply:
Sounds like something we need to fix in the test, not with a go routine here.